From 0277e7342945f917ff5817bf6712031cdc907e42 Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Fri, 18 Jan 2019 10:42:41 +0000
Subject: [PATCH 1/4] LineageSettingsActivity: Account for user stored
 KeyDisabler setting

 * This ensures the proper switch status is set
   when migrating from older versions

Change-Id: Ie6b79a5269418ec7eae205e30558449098202e54
---
 .../setupwizard/LineageSettingsActivity.java          | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/src/org/lineageos/setupwizard/LineageSettingsActivity.java b/src/org/lineageos/setupwizard/LineageSettingsActivity.java
index d72d2e0..985037d 100644
--- a/src/org/lineageos/setupwizard/LineageSettingsActivity.java
+++ b/src/org/lineageos/setupwizard/LineageSettingsActivity.java
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2016 The CyanogenMod Project
- * Copyright (C) 2017-2018 The LineageOS Project
+ * Copyright (C) 2017-2019 The LineageOS Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -127,8 +127,8 @@ public class LineageSettingsActivity extends BaseSetupWizardActivity {
         mNavKeys = (CheckBox) findViewById(R.id.nav_keys_checkbox);
         mSupportsKeyDisabler = isKeyDisablerSupported(this);
         if (mSupportsKeyDisabler) {
-            boolean navKeysDisabled = isKeyDisablerActive(this);
-            mNavKeys.setChecked(navKeysDisabled);
+            mNavKeys.setChecked(LineageSettings.System.getIntForUser(getContentResolver(),
+                    LineageSettings.System.FORCE_SHOW_NAVBAR, 0, UserHandle.USER_CURRENT) != 0);
         } else {
             navKeysRow.setVisibility(View.GONE);
         }
@@ -216,9 +216,4 @@ public class LineageSettingsActivity extends BaseSetupWizardActivity {
         final LineageHardwareManager hardware = LineageHardwareManager.getInstance(context);
         return hardware.isSupported(LineageHardwareManager.FEATURE_KEY_DISABLE);
     }
-
-    private static boolean isKeyDisablerActive(Context context) {
-        final LineageHardwareManager hardware = LineageHardwareManager.getInstance(context);
-        return hardware.get(LineageHardwareManager.FEATURE_KEY_DISABLE);
-    }
 }
-- 
2.17.1

